cmd/virtual-workspaces/command/cmd.go

```diff
@@ -74,49 +76,59 @@ func NewCommand(errout io.Writer, stopCh <-chan struct{}) *cobra.Command {
 }
 
 // Run takes the options, starts the API server and waits until stopCh is closed or initial listening fails.
-func Run(o *options.Options, stopCh <-chan struct{}) error {
+func Run(ctx context.Context, o *options.Options) error {
 	// parse kubeconfig
 	kubeConfig, err := readKubeConfig(o.KubeconfigFile)
 	if err != nil {
 		return err
 	}
-	kubeClientConfig, err := kubeConfig.ClientConfig()
+	nonIdentityConfig, err := kubeConfig.ClientConfig()
 	if err != nil {
 		return err
 	}
-	u, err := url.Parse(kubeClientConfig.Host)
+	u, err := url.Parse(nonIdentityConfig.Host)
 	if err != nil {
 		return err
 	}
 	u.Path = ""
-	kubeClientConfig.Host = u.String()
+	nonIdentityConfig.Host = u.String()
 
-	// create clients and informers
-	kubeClusterClient, err := kubernetes.NewClusterForConfig(kubeClientConfig)
+	// resolve identities for system APIBindings
+	nonIdentityKcpClusterClient, err := kcpclient.NewClusterForConfig(nonIdentityConfig) // can only used for apis.kco.dev
 	if err != nil {
 		return err
 	}
+	identityConfig, resolveIdentities := boostrap.NewConfigWithWildcardIdentities(nonIdentityConfig, boostrap.KcpRootGroupExportNames, boostrap.KcpRootGroupResourceExportNames, nonIdentityKcpClusterClient.Cluster(tenancyv1alpha1.RootCluster))
+	if err := wait.PollImmediateInfiniteWithContext(ctx, time.Millisecond*500, func(ctx context.Context) (bool, error) {
+		if err := resolveIdentities(ctx); err != nil {
+			klog.V(3).Infof("failed to resolve identities, keeping trying: %v", err)
+			return false, nil
+		}
+		return true, nil
+	}); err != nil {
+		klog.Errorf("failed to get or create identities: %v", err)
+		// nolint:nilerr
+		return nil // don't klog.Fatal. This only happens when context is cancelled.
```

Comment:
I feel like we could return the error here?

Reply:
done